src/util.test.js

```diff
@@ -64,4 +82,89 @@ describe('util', () => {
       })).toEqual('LastName, FirstName MiddleName');
     });
   });
+
+  describe('getCheckoutSettings', () => {
+    it('should return undefined if input length is 0', () => {
+      expect(getCheckoutSettings([])).toEqual(undefined);
+    });
+
+    it('should return parsed JSON for a valid input', () => {
+      expect(getCheckoutSettings(settings)).toEqual({ prefPatronIdentifier: 'barcode,id' });
+    });
+
+    it('should return an empty object if there\'s an error', () => {
+      expect(getCheckoutSettings(emptySettings)).toEqual({});
+    });
+  });
+
+  describe('getPatronIdentifiers', () => {
+    it('should return the identifiers from a valid prefs object', () => {
+      expect(getPatronIdentifiers(settings)).toContain('barcode', 'id');
+    });
+
+    it('should return default identifiers in case of invalid prefs object', () => {
+      expect(getPatronIdentifiers(emptySettings)).toContain(defaultPatronIdentifier);
+    });
+  });
+
+  describe('buildIdentifierQuery', () => {
+    it('should convert a set of identifiers to a CQL query', () => {
+      const idents = getPatronIdentifiers(settings);
+
+      expect(buildIdentifierQuery({ identifier: mockId }, idents))
+        .toEqual(`(${idents[0]}=="${mockId}" OR ${idents[1]}=="${mockId}")`);
+    });
+  });
+
+  describe('buildRequestQuery', () => {
+    it('should convert requesterId to a CQL query', () => {
+      expect(buildRequestQuery(mockId))
+        .toEqual(`(requesterId==${mockId} and status=="Open - Awaiting pickup")`);
+    });
+
+    it('should convert requesterId and servicePointId to a CQL query', () => {
+      expect(buildRequestQuery(mockId, mockId))
+        .toEqual(`(requesterId==${mockId} and pickupServicePointId=${mockId} and status=="Open - Awaiting pickup")`);
+    });
+  });
+
+  describe('getPatronBlocks', () => {
+    const manualBlock = { borrowing: true, expirationDate: '2101-01-01' };
+    const automatedBlock = { blockBorrowing: true };
+
+    it('should return patron blocks fulfilling the requirements', () => {
+      expect(getPatronBlocks([manualBlock], [automatedBlock]))
+        .toEqual([...[automatedBlock], ...[manualBlock]]);
+    });
+
+    it('should return empty array if patron blocks don\'t fulfill the requirements', () => {
+      expect(getPatronBlocks([{ ...manualBlock, borrowing: false }], []))
+        .toEqual([]);
+    });
+  });
+
+  describe('shouldStatusModalBeShown', () => {
+    it('should return true if item status is in the list', () => {
+      expect(shouldStatusModalBeShown({ status: { name: 'Unknown' } }))
+        .toEqual(true);
+    });
+
+    it('should return false if item status is not in the list', () => {
+      expect(shouldStatusModalBeShown({ status: { name: 'U-02' } })).toEqual(false);
+    });
+  });
+
+  describe('renderOrderedPatronBlocks', () => {
```

Comment:
Could you check all conditionals?
![image](https://user-images.githubusercontent.com/42437614/218497441-799eb3be-a1c8-45d7-b0c9-5abfc70b19d2.png)

Regarding `Raw` component you just can mock it and check that it was called with correct props.